diff --git b/build/libltdl/loaders/preopen.c a/build/libltdl/loaders/preopen.c
index 139e373..6e45a2f 100644
--- b/build/libltdl/loaders/preopen.c
+++ a/build/libltdl/loaders/preopen.c
@@ -282,7 +282,7 @@ add_symlist (const lt_dlsymlist *symlist)
 	  if (symlist[1].name && STREQ (symlist[1].name, "@INIT@"))
 	    {
 	      void (*init_symlist)(void);
-	      *(void **)(&init_symlist) = symlist[1].address;
+	      init_symlist = (void (*)(void))(symlist[1].address);
 	      (*init_symlist)();
 	    }
 	}
diff --git a/build/libltdl/ltdl.c b/build/libltdl/ltdl.c
index 6013f2ac2..0283e57fb 100644
--- a/build/libltdl/ltdl.c
+++ b/build/libltdl/ltdl.c
@@ -33,6 +33,10 @@ or obtained by writing to the Free Software Foundation, Inc.,
 #include "lt_system.h"
 #include "lt_dlloader.h"
 
+#if defined(__clang_major__) && __clang_major__ > 3
+#pragma clang diagnostic ignored "-Wconstant-logical-operand"
+#endif /* clang warnings */
+
 
 /* --- MANIFEST CONSTANTS --- */
 
